docs/aws/audit/dynamodbmonitoring/rules/ddb_table_autoscaling_enabled_remediation.mdx

```diff
@@ -1,6 +1,284 @@
+       ServiceNamespace='dynamodb',
+       ResourceId=resource_id
+   )
+   print(response)
+   ```
+
+4. **Automate the Process**:
+   To ensure that all new DynamoDB tables have autoscaling enabled, you can automate the table creation process using the above script. You can also integrate this script into your CI/CD pipeline to enforce autoscaling policies on all newly created tables.
+
+By following these steps, you can ensure that all DynamoDB tables created using your Python scripts have autoscaling enabled, thereby preventing misconfigurations related to autoscaling.
+</Accordion>
+
+</AccordionGroup>
+</Tab>
+<Tab title='Cause'>
+### Check Cause
+<AccordionGroup>
+<Accordion title='Using Console' defaultOpen='true'>
+1. Sign in to the AWS Management Console.
+2. Navigate to the DynamoDB service by typing 'DynamoDB' into the search bar and selecting it from the dropdown menu.
+3. In the DynamoDB dashboard, select 'Tables' from the left-hand navigation pane.
+4. Select the table you want to check for autoscaling. This will open the table details page.
+5. In the table details page, select the 'Capacity' tab. Here, you can see whether autoscaling is enabled or not. If 'Read capacity' and 'Write capacity' are set to 'Provisioned', and 'Auto Scaling' for both read and write capacities are enabled, then autoscaling is enabled for the table. If not, then autoscaling is not enabled.
+</Accordion>
+
+<Accordion title='Using CLI'>
+1. First, you need to list all the DynamoDB tables in your AWS account. You can do this by using the AWS CLI command `aws dynamodb list-tables`. This command will return a list of all the DynamoDB tables in your account.
+
+2. Once you have the list of tables, you can check the autoscaling settings for each table. This can be done using the AWS CLI command `aws application-autoscaling describe-scalable-targets --service-namespace dynamodb`. This command will return a list of all the scalable targets for DynamoDB in your account.
+
+3. From the output of the above command, you need to check the `ScalableDimension` field. If the value of this field is `dynamodb:table:ReadCapacityUnits` or `dynamodb:table:WriteCapacityUnits`, it means that autoscaling is enabled for read capacity or write capacity respectively.
+
+4. Finally, you need to check the `MinCapacity` and `MaxCapacity` fields. If the `MinCapacity` is not 0 or the `MaxCapacity` is not a high enough value, it means that autoscaling may not be properly configured.
+</Accordion>
+
+<Accordion title='Using Python'>
+1. Install and configure AWS SDK for Python (Boto3) on your local system. This will allow you to interact with AWS services including DynamoDB. You can install it using pip:
+
+```python
+pip install boto3
+```
+
+2. Import the necessary modules and establish a session with AWS using your access keys:
+
+```python
+import boto3
+
+session = boto3.Session(
+    aws_access_key_id='YOUR_ACCESS_KEY',
+    aws_secret_access_key='YOUR_SECRET_KEY',
```

Comment:
<b>Secret:</b> Secret Keyword <br/>
                    <b><a href="https://console.cloudanix.com/code?t=secrets" target='_blank'>Click here for more details</a></b>